riscv/src/lib.rs

```diff
@@ -124,6 +124,9 @@ pub fn compile_rust_to_riscv_asm(input_file: &str) -> BTreeMap<String, String> {
 name = "{}"
 version = "0.1.0"
 edition = "2021"
+
+[dependencies]
+runtime = {{ path = "../runtime" }}
```

Comment:
This is a way to do it, but the value has to be copied in the right directory. I think the better alternative is to publish this as a crate, but would like some input before I do so.

Reply:
To be clear, this will not work unless the runtime crate is manually copied to the right location. I can do that inside `compiler.rs`, but it feels tacky because it would have to be around during compilation. So my preferred approach is to have a published crate/github reference/path override option.

Reply:
I mean this case is only for single rust files, where we create a crate for them. If we use a published link, then any compilation needs network access, which is not so good in my opinion. So maybe we could create a "resource file" out of it and create the files accordingly? Mabe there is also a more "rust-y" approach.

Reply:
so what I did was to package the runtime code inside the riscv compiler executable, using `include_bytes!` and then make sure it's added to the crate that is created by the compiler for each file. This will save a download. For regular crate, a dependency can be given, but for the one in the example, the relative path is passed in its `Cargo.toml`.